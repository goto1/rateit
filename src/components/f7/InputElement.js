import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Icon from "./Icon";
import pick from "lodash/pick";

const InputWrapper = styled.div`margin-left: 15px;`;

const InputElement = props => {
  const { icon } = props;
  const inputAttrs = pick(props, [
    "id",
    "name",
    "type",
    "value",
    "placeholder",
    "onChange"
  ]);
  return (
    <div className="item-content">
      <div className="item-media">
        <Icon material={icon} />
      </div>
      <InputWrapper className="item-input">
        <input {...inputAttrs} />
      </InputWrapper>
    </div>
  );
};

InputElement.propTypes = {
  icon: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  id: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func
};

export default InputElement;
