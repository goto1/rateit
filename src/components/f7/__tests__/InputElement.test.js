import React from "react";
import InputElement from "../InputElement";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

const props = {
  icon: "account",
  type: "text",
  name: "user_name",
  id: "user_name",
  placeholder: "Your name goes here..",
  value: "Jim",
  onChange: jest.fn()
};

describe("InputElement component", () => {
  it("should renders correctly", () => {
    const tree = renderer.create(<InputElement {...props} />).toJSON();
    expect(tree).toMatchStyledComponentsSnapshot();
  });

  it("should render the input element with specified attributes", () => {
    const wrapper = shallow(<InputElement {...props} />);
    const actual = wrapper.find("input").props().name;
    const expected = "user_name";
    expect(actual).toBe(expected);
  });

  it("should call a function when input changes", () => {
    const wrapper = shallow(<InputElement {...props} />);
    wrapper.find("input").simulate("change");
    const actual = props.onChange.mock.calls.length;
    const expected = 1;
    expect(actual).toBe(expected);
  });
});
