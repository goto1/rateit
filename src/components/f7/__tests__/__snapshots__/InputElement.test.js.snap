// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputElement component should renders correctly 1`] = `
.bSLHOK {
  margin-left: 15px;
}

<div
  className="item-content"
>
  <div
    className="item-media"
  >
    <i
      className="material-icons icon"
    >
      account
    </i>
  </div>
  <div
    className="item-input sc-bdVaJa bSLHOK"
  >
    <input
      id="user_name"
      name="user_name"
      onChange={[Function]}
      placeholder="Your name goes here.."
      type="text"
      value="Jim"
    />
  </div>
</div>
`;
